cpp/src/unary/cast_ops.cu

```diff
@@ -236,7 +269,56 @@ struct dispatch_unary_cast_to {
     return output;
   }
 
+  template <
+    typename TargetT,
+    typename SourceT                                                  = _SourceT,
+    typename std::enable_if_t<cudf::is_fixed_point<SourceT>() && cudf::is_fixed_point<TargetT>() &&
+                              std::is_same<SourceT, TargetT>::value>* = nullptr>
+  std::unique_ptr<column> operator()(data_type type,
+                                     rmm::mr::device_memory_resource* mr,
+                                     cudaStream_t stream)
+  {
+    if (input.type() == type) return std::make_unique<column>(input);  // TODO add test for this
+
+    return detail::rescale<TargetT>(input, type, mr, stream);
+  }
+
+  template <
+    typename TargetT,
+    typename SourceT                                                      = _SourceT,
+    typename std::enable_if_t<cudf::is_fixed_point<SourceT>() && cudf::is_fixed_point<TargetT>() &&
+                              not std::is_same<SourceT, TargetT>::value>* = nullptr>
+  std::unique_ptr<column> operator()(data_type type,
+                                     rmm::mr::device_memory_resource* mr,
+                                     cudaStream_t stream)
+  {
+    using namespace numeric;
+
+    auto const size = input.size();
+    auto temporary =
+      std::make_unique<column>(cudf::data_type{type.id(), input.type().scale()},
+                               size,
+                               rmm::device_buffer{size * cudf::size_of(type), stream, mr},
+                               copy_bitmask(input, stream, mr),
```

Comment:
using `mr` for temporary? it should be `rmm::mr::get_current_device_resource()`.

Reply:
I fine with making the change but I've never heard about this. Can you explain the rationale?

Reply:
This is fundamental to every libcudf API since the advent of libcudf++. See the docs for any MR parameter: they all say "to use for allocating the return/output" or something similar. Temporary values used internally should be allocated using the default resource.  See https://github.com/rapidsai/cudf/blob/branch-0.17/cpp/docs/TRANSITIONGUIDE.md#temporary-memory